packages/hdwallet-native/src/crypto/isolation/adapters/ethereum.ts

```diff
@@ -32,13 +35,15 @@ export class SignerAdapter extends ethers.Signer {
     throw new Error("changing providers on a SignerAdapter is unsupported");
   }
 
-  async getAddress(): Promise<string> {
-    return ethers.utils.computeAddress(SecP256K1.UncompressedPoint.from(await this._isolatedKey.getPublicKey()));
+  async getAddress(addressNList: core.BIP32Path): Promise<string> {
+    const addressNode = await this.rootNode.derivePath(core.addressNListToBIP32(addressNList));
```

Comment:
`addressNode` -> `node` might be a bit better vernacular. No reason these nodes couldn't be derived at any level of the path and just more generic

Reply:
`rootNode` was definitely the wrong name. I changed it to master which I think reads better and better reflects what the variable is. lmk

Reply:
parent/child vernacular could be solid as well, parent is passed through (may be root or lower depth) and a child node is derived via the provided path

Reply:
in the end @pastaghost wins the naming contest with `nodeAdapter`